### PR TITLE
Token provider testing

### PR DESCRIPTION
**Pull request by iainmcgin**
Adds the ability to set token provider request properties for
hint and credential retrieve requests. Renders raw ID token
response (placeholder for better rendering).

### CODE REVIEW

File: testapp/java/org/openyolo/testapp/HintTestPageFragment.java [hunk 1]
```diff
@@ -83,6 +89,9 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
         Hint hint = result.getHint();
         if (hint != null) {
             mCredentialView.setFieldsFromHint(hint);
+            if (hint.getIdToken() != null) {
```
Comment (dxslly) [inline testapp/java/org/openyolo/testapp/HintTestPageFragment.java]: No need to null check, setIdToken accepts nullable values

File: testapp/java/org/openyolo/testapp/RetrieveTestPageFragment.java [hunk 2]
```diff
@@ -83,8 +90,12 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
         }
 
         CredentialRetrieveResult result = mApi.getCredentialRetrieveResult(data);
-        if (result.getCredential() != null) {
-            mCredentialView.setFieldsFromCredential(result.getCredential());
+        Credential credential = result.getCredential();
+        if (credential != null) {
+            mCredentialView.setFieldsFromCredential(credential);
+            if (credential.getIdToken() != null) {
```
Comment (dxslly) [inline testapp/java/org/openyolo/testapp/RetrieveTestPageFragment.java]: Same here

File: testapp/java/org/openyolo/testapp/TokenProviderInputView.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    public Map<String, TokenRequestInfo> getTokenProviders() {
+        HashMap<String, TokenRequestInfo> tokenProviders = new HashMap<>();
+        String tokenProviderUriStr = mTokenProviderUriField.getText().toString().trim();
+
+        if (tokenProviderUriStr.isEmpty()) {
+            return tokenProviders;
+        }
+
+        Uri tokenProviderUri = Uri.parse(tokenProviderUriStr);
+        if (!"https".equals(tokenProviderUri.getScheme())) {
```
Comment (dxslly) [inline testapp/java/org/openyolo/testapp/TokenProviderInputView.java]: What's leading you to add this constraint?

Comment (iainmcgin) [inline testapp/java/org/openyolo/testapp/TokenProviderInputView.java]: I thought I had spec'd it that way, apparently not. Will remove.

File: testapp/java/org/openyolo/testapp/HintTestPageFragment.java (hunk 1 above)
Comment (iainmcgin) [inline testapp/java/org/openyolo/testapp/HintTestPageFragment.java]: Done

File: testapp/java/org/openyolo/testapp/RetrieveTestPageFragment.java (hunk 2 above)
Comment (iainmcgin) [inline testapp/java/org/openyolo/testapp/RetrieveTestPageFragment.java]: Done

**Pull request merged**